src/lib.rs

```diff
@@ -1159,25 +1151,51 @@ impl StoredPath {
 
     /// Returns the final component of the path, if there is one.
     pub fn file_name(&self) -> Option<&str> {
+        let mut path = self.as_str();
```

Comment:
Sorry, I am coming into this a little cold. Why is a custom implementation of finding a file name needed? All of the functionality around `StoredPath` really feels like it should its own crate/package or might already be provided by an existing crate/package. I just know there be dragons here.

Reply:
Maybe to start we can move all of the stored path code to a separate module from the `lib.rs` module?

Reply:
I agree it's increasingly wanting to be a separate crate -- the reason a separate impl is needed is because std::path/camino both have host-platform-specific behaviours on paths, and there's no way to ask for anything else. 

Most notably, `\` is a valid character for a unix filename, so if you're on linux `a/b\c.txt` will interpretted as `filename = "b\c.txt"`, which is not desirable for us.